lib/models/device.coffee

```diff
@@ -291,16 +227,9 @@ exports.has = (name, callback) ->
 #		console.log("Is device online? #{isOnline}")
 ###
 exports.isOnline = (name, callback) ->
-
-	if not callback?
-		throw new errors.ResinMissingParameter('callback')
-
-	if not _.isFunction(callback)
-		throw new errors.ResinInvalidParameter('callback', callback, 'not a function')
-
-	exports.get name, (error, device) ->
-		return callback(error) if error?
-		return callback(null, !!device.is_online)
+	exports.get(name).then (device) ->
+		return !!device.is_online
```

Comment:
It should always give `true`/`false` (and the check I just did showed `true`/`false`), I did fix a bug ages ago where it would return `0`/`1` in a $expand though, which might be what you experienced on the ui at some point - if do you find any other cases where it happens then let me know